pkg/tenant/members.go

```diff
@@ -467,3 +502,42 @@ func (s *Server) InvitePreview(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, out)
 }
+
+// Helper method to check if an organization has at least one owner.
+func hasOrgOwner(c *gin.Context, role string, orgID ulid.ULID) error {
+	var err error
+
+	// If the member to be updated is an owner, loop over dbMember and break out of the loop if there are at least two owners.
+	// If member is the only owner, their role cannot be changed.
+	if role == perms.RoleOwner {
+		// Get members from the database and set page size to return all members.
+		// TODO: Create list method that will not require pagination for this endpoint.
+		getAll := &pg.Cursor{StartIndex: "", EndIndex: "", PageSize: 100}
+		var members []*db.Member
+		if members, _, err = db.ListMembers(c, orgID, getAll); err != nil {
+			sentry.Error(c).Err(err).Msg("could not list members")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not get member from the database"))
```

Comment:
To simplify this further, I would prefer if we would not do any logging or JSON responses in this method (save it for the handler). Perhaps this method can just return the number of owners and we can have the `switch count` on the handler itself?